### PR TITLE
feat(make:factory): match directory/namespace structure

### LINKED ISSUE #401: Factory Maker: match directory/namespace structure
Hi, it would be nice if the `make:factory` command could match the entity namespace/directory structure?

Eg. for an entity with the `App\Entity\Franchise\Shop` FQCN, it would create a factory in the `src/Factory/Franchise/Shop.php` file, in the `App\Factory\Franchise` namespace.

### PR DESCRIPTION
**Pull request by nikophil**
fix #401 

I've also added a configuration to chose the default namespace that will be used

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -204,6 +204,24 @@ This command will generate a ``PostFactory`` class that looks like this:
...
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/packages/zenstruck_foundry.yaml
+            when@dev: # see Bundle Configuration section about sharing this in the test environment
+                zenstruck_foundry:
+                    maker:
+                        factory:
+                            default_namespace: 'App\\MyFactories'
```
Comment (nikophil) [inline docs/index.rst]: I'm wondering if this is not too "deep" just for one conf, but it seemed semantically more correct

File: docs/index.rst [hunk 2]
```diff
@@ -2015,8 +2042,19 @@ Full Default Bundle Configuration
                     'object_managers' => false,
                 ],
 
-                // Add global state
-                'global_state' => []
```
Comment (nikophil) [inline docs/index.rst:2019]: there was an error in the doc: `global_state` should be a direct child of `zenstruck_foundry`

File: docs/index.rst [hunk 3]
```diff
@@ -204,6 +204,24 @@ This command will generate a ``PostFactory`` class that looks like this:
 
     Using ``make:factory --test`` will generate the factory in ``tests/Factory``.
 
+.. tip::
+
+    You can configure globally it which namespace the factories will be generated:
```
Comment (kbond) [inline docs/index.rst]:
```suggestion
    You can globally configure which namespace the factories will be generated in:
```

File: docs/index.rst [hunk 4]
```diff
@@ -204,6 +204,24 @@ This command will generate a ``PostFactory`` class that looks like this:
...
+        .. code-block:: yaml
+
+            # config/packages/zenstruck_foundry.yaml
+            when@dev: # see Bundle Configuration section about sharing this in the test environment
+                zenstruck_foundry:
+                    maker:
+                        factory:
+                            default_namespace: 'App\\MyFactories'
+
+    You can override this configuration by using the option ``--namespace``.
```
Comment (kbond) [inline docs/index.rst]:
```suggestion
    You can override this configuration by using the ``--namespace`` option.
```

File: docs/index.rst (hunk 1 above)
Comment (kbond) [reply docs/index.rst]: I don't have a strong opinion but what about:

```yaml
zenstruck_foundry:
    make_factory:
        default_namespace: ...
```

**Review by kbond — CHANGES_REQUESTED** [review]
Looks good, just a few minor things.

**Review by kbond — APPROVED** [review]
Beautiful!

**Pull request merged**